staging/src/k8s.io/client-go/examples/create-update-delete-deployment/main.go

```diff
@@ -61,6 +65,9 @@ func main() {
 					Labels: map[string]string{
 						"app": "demo",
 					},
+					Annotations: map[string]string{
```

Comment:
not sure why you needed to add an annotation. :) this example was already designed to change replica count (visible in kubectl get/describe, too)

Reply:
I think it's to demonstrate the rollback? Annotations is part of the podTemplate, changing templates results in a new revision, so that the deployment can be rolled back.

Reply:
mainly to make it more obvious that the rollback occurred- replica count will not change after rollback. But you're right, it's not really necessary, if it causes confusion I can remove it :)

Reply:
> so that the deployment can be rolled back.

I think the replicaCount also creates a new Deployment revision even though it's not part of PodTemplate?

But yeah it's good to demonstrate pod rollout as well, but right now adding a `fizz=buzz` annotation and not explaining doesn't make it clear to the reader why we're updating two things.

Reply:
As far as I know, only changing replicaCount will not trigger a rollout
> Note: A Deployment’s revision is created when a Deployment’s rollout is triggered. This means that the new revision is created if and only if the Deployment’s pod template (.spec.template) is changed, for example if you update the labels or container images of the template.

[See here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)

Reply:
Instead of changing annotations, how about changing to a newer/older version of the [nginx](https://github.com/jekohk/kubernetes/blob/ccfdd2ae7bfebdc804da6fb30875906e97ac106d/staging/src/k8s.io/client-go/examples/create-update-delete-deployment/main.go#L76) image? That's a more realistic use case as well.

Reply:
agreed, changing the image is a more realistic use case & better illustrates how rollbacks work.

Reply:
+1